src/App.js

```diff
@@ -112,35 +119,65 @@ class App extends React.Component {
         let dataCards = this.state.dataCard.dataByStatuses;
         let card = null;
                 
-        Object.values(dataCards).find(array => (
+        dataCards && Object.values(dataCards).find(array => (
             card = array.find(item => item._id === id)             
         ));
         
         return card;
     }
-
-    render() {
+    
+    MainPage = () => {
         const {statuses, dataByStatuses} = this.state.dataCard;  
-        const {idCard} = this.state;      
-        
+        const {idCard} = this.state;
+
         return (
             <>
-            <div>
-                <CardAddPanel onCreateCard={this.handleCreateCard} />
-
-                <div class = "flex-row">
-                    {statuses && statuses.map(status => ( 
-                        <Section 
-                            status={status} 
-                            cards={dataByStatuses[status] || []}
-                            onDeleteCard={this.handleDeleteCard}
-                            onModalInfo={this.handleModalInfo}
-                        />)) }
-                </div>                
-            </div> 
+            <CardAddPanel onCreateCard={this.handleCreateCard} />
+
+            <div class = "flex-row">
+                {statuses && statuses.map(status => ( 
+                    <Section 
+                        status={status} 
+                        cards={dataByStatuses[status] || []}                        
+                        onDeleteCard={this.handleDeleteCard}
+                        onModalInfo={this.handleModalInfo}
+                    />)) }
+            </div>                           
             
-            {idCard && <ModalCard onCloseModal={this.handleCloseModal} {...this.findCardById(idCard)} />}
-            </>   
+            {idCard && 
+            <ModalCard onCloseModal={this.handleCloseModal} {...this.findCardById(idCard)}>
+                { card => (
+                    <CardInfo {...card} />
+                )}
+            </ModalCard>} 
+            </>
+        )
+    }
+
+    CardPage = () => {
+        let {id} = useParams();
+        let card = this.findCardById(id);        
+
+        return (
+            card && 
+                <div>
+                    <CardInfo {...card} />                    
+                </div>                
+        )
+    }
+
+    render() {       
+        return (
+            <Router>
+                <Switch>                             
+                    <Route exact path='/'>
+                        <this.MainPage />                                  
+                    </Route>
+                    <Route path='/:id'>                       
+                        <this.CardPage />                  
```

Comment:
please use Route like in docs
```
<Route path="/">
     <Home />
</Route>
```
you don't need methods, separate to components and use